auth.go

```diff
@@ -403,24 +411,27 @@ func (x *Central) GetTokenFromIdentityPassword(identity, password string) (*Toke
 		x.Stats.IncrementEmptyIdentities(x.Log)
 		return nil, ErrIdentityEmpty
 	}
-	userId, identity, eAuth := x.authenticate(identity, password)
+	user, authMsg, eAuth := x.authenticate(identity, password)
 	if eAuth == nil {
-		if permit, ePermit := x.permitDB.GetPermit(userId); ePermit == nil {
+		if permit, ePermit := x.permitDB.GetPermit(user.UserId); ePermit == nil {
 			t := &Token{}
 			t.Expires = veryFarFuture
-			t.Identity = identity
-			t.UserId = userId
+			t.Identity = user.getIdentity()
+			t.UserId = user.UserId
 			t.Permit = *permit
 			x.Stats.IncrementGoodOnceOffAuth(x.Log)
-			x.Log.Infof("Once-off auth successful (%v)", userId)
+			x.Log.Infof("Once-off auth successful (%v)", user.UserId)
+			if authMsg != "" {
+				t.Message = authMsg
+			}
```

Comment:
Just `t.Message = authMsg` because it will do exactly the same. 

And you can shove it in at `t`s creation time like i mentioned above.